compliance/instance_types/list_new_instance_types.pt

```diff
@@ -0,0 +1,395 @@
+  end
+end
+
+pagination "google_pagination" do
+  get_page_marker do
+    body_path "nextPageToken"
+  end
+  set_page_marker do
+    query "pageToken"
+  end
+end
+
+###############################################################################
+# Datasource
+###############################################################################
+
+datasource "ds_github_aws_instance_types" do
+  request do
+    verb "GET"
+      host "raw.githubusercontent.com"
+      path "/rightscale/policy_templates/master/data/aws/instance_types.json"
+      header "User-Agent","RS Policies"
+    end
+end
+
+datasource "ds_github_azure_instance_types" do
+  request do
+    verb "GET"
+      host "raw.githubusercontent.com"
+      path "/rightscale/policy_templates/master/data/azure/instance_types.json"
+      header "User-Agent","RS Policies"
+    end
+end
+
+datasource "ds_github_google_instance_types" do
+  request do
+    verb "GET"
+      host "raw.githubusercontent.com"
+      path "/rightscale/policy_templates/master/data/google/instance_types.json"
+      header "User-Agent","RS Policies"
+    end
+end
+
+datasource "ds_aws_instance_types" do
+  request do
+    auth $auth_aws
+    verb "GET"
+    host "ec2.amazonaws.com"
+    path "/"
+    query "Action", "DescribeInstanceTypes"
+    query "Version", "2016-11-15"
+    pagination $aws_pagination
+  end
+  result do
+    encoding "xml"
+    collect xpath(response, "//DescribeInstanceTypesResponse/instanceTypeSet/item", "array") do
+      field "instanceType", xpath(col_item, "instanceType")
+    end
+  end
+end
+
+datasource "ds_subscriptions" do
+  request do
+    auth $azure_auth
+    host "management.azure.com"
+    path "/subscriptions/"
+    query "api-version","2019-06-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "subscriptionId", jmes_path(col_item,"subscriptionId")
+    end
+  end
+end
+
+datasource "ds_filter_one_subscription" do
+  run_script $js_filter_one_subscription, $ds_subscriptions
+end
+
+datasource "ds_azure_locations" do
+  iterate $ds_filter_one_subscription
+  request do
+    auth $azure_auth
+    host "management.azure.com"
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"), "/locations"])
+    query "api-version","2020-01-01"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "locationName", jmes_path(col_item,"name")
+      field "displayName", jmes_path(col_item, "displayName")
+      field "subscriptionId",val(iter_item,"subscriptionId")
+    end
+  end
+end
+
+datasource "ds_azure_instance_types" do
+  iterate $ds_azure_locations
+  request do
+    auth $azure_auth
+    host "management.azure.com"
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"), "/providers/Microsoft.Compute/locations/", val(iter_item,"locationName"),"/vmSizes"])
+    query "api-version","2020-06-01"
+    ignore_status [400]
```

Comment:
Any particular reason to ignore the 400 status code

Reply:
We were getting 400 error so, I have added ignore status 400.

Error Message:
{"error":{"code":"NoRegisteredProviderFound","message":"No registered resource provider found for location 'centralusstage' and API version '2020-06-01' for type 'locations/vmSizes'. The supported api-versions are '2015-05-01-preview, 2015-06-15, 2016-03-30, 2016-04-30-preview, 2016-08-30, 2017-03-30, 2017-12-01, 2018-04-01, 2018-06-01, 2018-10-01, 2019-03-01, 2019-07-01, 2019-12-01, 2020-06-01, 2020-12-01'. The supported locations are 'eastus, eastus2, westus, centralus, northcentralus, southcentralus, northeurope, westeurope, eastasia, southeastasia, japaneast, japanwest, australiaeast, australiasoutheast, australiacentral, brazilsouth, southindia, centralindia, westindia, canadacentral, canadaeast, westus2, westcentralus, uksouth, ukwest, koreacentral, koreasouth, francecentral, southafricanorth, uaenorth, switzerlandnorth, germanywestcentral, norwayeast'."}}